DTML.EduBot/Controllers/MessagesController.cs

```diff
@@ -22,13 +23,13 @@ public class MessagesController : ApiController
             }
             else
             {
-                HandleSystemMessage(activity);
+                await HandleSystemMessage(activity);
```

Comment:
Async methods would generally have an async suffix. Can you rename HandleSystemMessage to HandleSystemMessageAsync?